src/planner/binder/statement/bind_update.cpp

```diff
@@ -105,6 +119,11 @@ static void BindUpdateConstraints(TableCatalogEntry &table, LogicalGet &get, Log
 		}
 	}
 
+	// if there is a self referencing foreign key constraint, update_is_del_and_insert sets to false
+	if (is_must_update) {
+		update.update_is_del_and_insert = false;
```

Comment:
I'm still not convinced this is correct - `update_is_del_and_insert` is set to true only when columns are updated that are contained in indexes, and **needs to** be set to true in this case since indexes do not correctly handle updates to individual columns.

Setting this to false when there are columns contained in the index being updated will lead to the index containing outdated or incorrect information, which will lead to wrong query results when the index is used.

Could you add tests that use the index after this condition is being triggered, e.g. through an equality predicate on the indexed column (e.g. `SELECT * FROM tbl WHERE indexed_column=1`).